webapp/src/test/java/com/minh/project/pages/GameTest.java

```diff
@@ -0,0 +1,199 @@
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+import java.time.Duration;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import com.minh.project.config.SeleniumWebDriverConfig;
+
+public class GameTest {
+	private static final String BASE_URL =  "http://localhost:8080/GameRate/";
+	
+    private SeleniumWebDriverConfig seleniumConfig;
+    private WebDriver driver;
+    private WebDriverWait wait;
+    
+    @FindBy(id="gameForm")
+	private WebElement gameForm;
+	
+	@FindBy(id="title")
+	private WebElement title;
+	
+	@FindBy(id="comment")
+	private WebElement comment;
+	
+	@FindBy(id="rating")
+	private WebElement rating;
+	
+	@FindBy(id="submitButton")
+	private WebElement submitButton;
+	
+    @FindBy(id="list")
+   	private WebElement list;
+    
+    @FindBy(id="gameId")
+  	private WebElement gameId;
+    
+    @FindBy(id="gameTitle")
+	private WebElement gameTitle;
+	
+	@FindBy(id="gameComment")
+	private WebElement gameComment;
+	
+	@FindBy(id="gameRating")
+	private WebElement gameRating;
+	
+	@FindBy(id="resetButton")
+	private WebElement resetButton;
+	
+	@FindBy(id="editButton")
+	private WebElement editButton;
+	
+	@FindBy(id="deleteButton")
+	private WebElement deleteButton;
+	
+	@BeforeSuite
+	public void setup() {
+		seleniumConfig = new SeleniumWebDriverConfig();
+		driver = seleniumConfig.getDriver();
+		PageFactory.initElements(driver, this); //required for selenium to find webelements
+	}
+	
+	@AfterSuite
+	public void tearDown() {
+		driver.close();
+		driver.quit();
+	}
+	
+	@BeforeTest
+	public void beforeEach() {
+		driver.navigate().to(BASE_URL + "game");
+		wait = new WebDriverWait(driver, Duration.ofSeconds(500));
+	}
+	
+	@Test
+	public void givenGamePage_whenAllInputFieldsValid_thenAddGameToList() {
+		enterFormData("Hollow Knight", "good music", 5);
+		assertTrue(isElementEnabled("submitButton", submitButton));
+		submit();
+		String expectedUrl = BASE_URL + "game";
+		String currentUrl = driver.getCurrentUrl();
+		assertEquals(currentUrl, expectedUrl, "Current URL is not the same as expected URL");
+		
+		wait.until(ExpectedConditions.presenceOfElementLocated(By.id("list")));
+		assertTrue(list.isDisplayed(), "Saved game does not appear on list");
+		assertTrue(gameTitle.getText().equals("Hollow Knight"), "Saved game does not have correct title");
+		assertFalse(isElementEnabled("submitButton", submitButton));
+		deleteButton.click();
+	}
+	
+	@Test
+	public void givenGamePage_whenEmptyTitle_thenAddButtonIsDisabled() {
```

Comment:
I like the consistency of your test method names. The given, when, then naming convention.